x-pack/plugins/ml/public/jobs/new_job/utils/new_job_utils.js

```diff
@@ -40,53 +41,58 @@ export function getQueryFromSavedSearch(formConfig) {
   };
 }
 
-// create items used for searching and job creation.
-// takes the $route object to retrieve the indexPattern and savedSearch from the url
-export function createSearchItems($route) {
-  let indexPattern = $route.current.locals.indexPattern;
-  const query = {
-    query_string: {
-      analyze_wildcard: true,
-      query: '*'
-    }
-  };
+// Provider for creating the items used for searching and job creation.
+// Takes the $route object to retrieve the indexPattern and savedSearch from the url
+export function SearchItemsProvider(Private) {
+
+  const buildESQuery = Private(BuildESQueryProvider);
 
-  let filters = [];
-  const savedSearch = $route.current.locals.savedSearch;
-  const searchSource = savedSearch.searchSource;
-
-  if (indexPattern.id === undefined &&
-    savedSearch.id !== undefined) {
-    indexPattern = searchSource.getField('index');
-
-    // Extract the query from the searchSource
-    // Might be as a String in q.query, or
-    // nested inside q.query.query_string
-    const q = searchSource.getField('query');
-    if (q !== undefined && q.language === 'lucene' && q.query !== undefined) {
-      if (typeof q.query === 'string' && q.query !== '') {
-        query.query_string.query = q.query;
-      } else if (typeof q.query === 'object' &&
-          typeof q.query.query_string === 'object' && q.query.query_string.query !== '') {
-        query.query_string.query = q.query.query_string.query;
+  function createSearchItemsFromRoute($route) {
```

Comment:
as this function is now wrapped in a provider, `$route` doesn't need to be passed in here and can be added to providers parameters above.
The function's name change may also then not be necessary, it could go back to being `createSearchItems`